classes/SpecificPrice.php

```diff
@@ -348,14 +438,30 @@ public static function getSpecificPrice($id_product, $id_shop, $id_currency, $id
         ** The price must not change between the top and the bottom of the page
         */
 
+        if (!self::couldHaveSpecificPrice($id_product)) {
+            return array();
+        }
+
         static $psQtyDiscountOnCombination = null;
         if ($psQtyDiscountOnCombination === null) {
             $psQtyDiscountOnCombination = Configuration::get('PS_QTY_DISCOUNT_ON_COMBINATION');
+            // no need to compute the key the first time the function is called, we know the cache has not
+            // been computed yet
+            $key = null;
+        } else {
+            $key = self::computeKey($id_product, $id_shop, $id_currency, $id_country, $id_group, $quantity,
+                $id_product_attribute, $id_customer, $id_cart,
+                $real_quantity);
```

Comment:
Please break your argument list so that there's only one argument per line.

According to PSR-2:
> Argument lists MAY be split across multiple lines, where each subsequent line is indented once. When doing so, the first item in the list MUST be on the next line, and there MUST be only one argument per line.

```php
<?php
$foo->bar(
    $longArgument,
    $longerArgument,
    $muchLongerArgument
);
```